apps/aevm/lib/state.ex

```diff
@@ -0,0 +1,285 @@
+        :destination => destination,
+        :gasLimit => gas_limit,
+        :value => value
+      }
+      | callcreates
+    ])
+  end
+
+  def set_stack(stack, state) do
+    Map.put(state, :stack, stack)
+  end
+
+  def set_memory(memory, state) do
+    Map.put(state, :memory, memory)
+  end
+
+  def set_storage(storage, state) do
+    Map.put(state, :storage, storage)
+  end
+
+  def set_pc(pc, state) do
+    Map.put(state, :pc, pc)
+  end
+
+  def set_out(out, state) do
+    Map.put(state, :out, out)
+  end
+
+  def set_logs(logs, state) do
+    Map.put(state, :logs, logs)
+  end
+
+  def set_gas(gas, state) do
+    Map.put(state, :gas, gas)
+  end
+
+  def set_selfdestruct(value, state) do
+    Map.put_new(state, :selfdestruct, value)
+  end
+
+  def stack(state) do
+    Map.get(state, :stack)
+  end
+
+  def memory(state) do
+    Map.get(state, :memory)
+  end
+
+  def storage(state) do
+    Map.get(state, :storage)
+  end
+
+  def code(state) do
+    Map.get(state, :code)
+  end
+
+  def pc(state) do
+    Map.get(state, :pc)
+  end
+
+  def jumpdests(state) do
+    Map.get(state, :jumpdests)
+  end
+
+  def logs(state) do
+    Map.get(state, :logs)
+  end
+
+  def address(state) do
+    Map.get(state, :address)
+  end
+
+  def caller(state) do
+    Map.get(state, :caller)
+  end
+
+  def data(state) do
+    Map.get(state, :data)
+  end
+
+  def gas(state) do
+    Map.get(state, :gas)
+  end
+
+  def gas_price(state) do
+    Map.get(state, :gasPrice)
+  end
+
+  def origin(state) do
+    Map.get(state, :origin)
+  end
+
+  def value(state) do
+    Map.get(state, :value)
+  end
+
+  def current_coinbase(state) do
+    Map.get(state, :currentCoinbase)
+  end
+
+  def current_difficulty(state) do
+    Map.get(state, :currentDifficulty)
+  end
+
+  def current_gas_limit(state) do
+    Map.get(state, :currentGasLimit)
+  end
+
+  def current_number(state) do
+    Map.get(state, :currentNumber)
+  end
+
+  def current_timestamp(state) do
+    Map.get(state, :currentTimestamp)
+  end
+
+  def get_balance(address, state) do
+    pre = Map.get(state, :pre)
+    account = Map.get(pre, address, %{})
+    Map.get(account, :balance, 0)
+  end
+
+  def get_ext_code_size(address, state) do
+    pre = Map.get(state, :pre)
+    account = Map.get(pre, address, %{})
+    code = Map.get(account, :code, <<>>)
+
+    byte_size(code)
+  end
+
+  def get_code(address, state) do
+    pre = Map.get(state, :pre)
+    account = Map.get(pre, address, %{})
+
+    Map.get(account, :code, <<>>)
+  end
+
+  def return_data(state) do
+    Map.get(state, :return_data)
+  end
+
+  def inc_pc(state) do
+    pc = Map.get(state, :pc)
+    Map.put(state, :pc, pc + 1)
+  end
+
+  def calculate_blockhash(nth_block, _a, state) do
+    # TODO: h -> header, maybe needs refactoring for an actual blockchain
+    current_number = current_number(state)
+
+    cond do
+      nth_block >= current_number ->
+        0
+
+      _a == 256 ->
+        0
+
+      # h == 0 -> 0
```

Comment:
Should this be removed ?

Reply:
This is an actual case but will be implemented in some other iteration